README.md

```diff
@@ -1,10 +1,43 @@
 # Heimdall
 
-**TODO: Add description**
+Heimdall is an API Gateway for Mesos and Marathon. It routes requests through
+filters dynamically loaded from Marathon before forwarding them
+to their intended services.
+
+## Usage
+
+When the application starts, it will attempt to register itself as a 
+Marathon subscriber. Once registered as a callback, Heimdall will start 
+listening for change events from Marathon, reloading its dynamic 
+routing config on each change.
+
+Dynamic routing configuration is done through Marathon labels. There are 4 
+labels used to decide how a request gets routed for an app:
+ * `heimdall.host` (required) - routes a request only if its host matches this value
+ * `heimdall.path` (required) - path prefix the request must match (the prefix is stripped from the forwarded request)
+ * `heimdall.filters` - a JSON list of plugs through which to filter the request before sending it back off
+ * `heimdall.opts` - a JSON object of options that gets passed into each plug
+
+Filters can be viewed as a pipeline of plugs (simple functions that take a 
+request and return a changed request or a response), 
+with the final plug forwarding the request to the place it actually needs
+ to go. The pipeline is built in the order that the plugs are listed in the
+config. In other words, if `heimdal.filters` is: `["Plugs.First", "Plugs.Second"]`
+ the pipeline would have the following flow:
+
+`Request -> Plugs.First -> Plugs.Second -> Heimdall.Plug.ForwardRequest -> Microservice`
+
+Ideally, Heimdall should forward requests to a load balancer that knows how to
+send each request to an actual instance of the microservice it's trying to get
+to. Some recommended solutions for this are [traefik](https://github.com/containous/traefik), [Minuteman](https://github.com/dcos/minuteman), or your own
+custom nginx config. The default forward url (ie the location of the load
+balancer) is an OTP application setting, `:default_forward_url`. If you
```

Comment:
for us erlang newbs maybe a link to an example of how to set otp settings would be nice